clamav/util.py

```diff
@@ -70,32 +71,47 @@ def scan_stream(fileobj):
     return status, signature_or_none
 
 
-def scan_container_image(image_reference: str):
-    logger.debug(f'scanning container image {image_reference}')
+def yield_files_from_container_image(
+    container_image_reference: str,
+) -> typing.Tuple[typing.IO, str]:
+    '''Yields tuples of file and path from the container image
 
+    The paths yielded are relative paths from inside the container image.
+    '''
     with tarfile.open(
         mode='r|',
-        fileobj=container.registry.retrieve_container_image(image_reference)
-    ) as tf:
-        for ti in tf:
+        fileobj=container.registry.retrieve_container_image(container_image_reference)
+    ) as tar_file:
+        for file_object in tar_file:
```

Comment:
NO! please read the documentation on `tarfile` first. the object returned is a `TarInfo` object. _NOT_ a FileObject. this is misleading!